src/Stratis.Bitcoin/Base/BaseFeature.cs

```diff
@@ -175,7 +175,6 @@ public override void Start()
 
             this.disposableResources.Add(this.timeSyncBehaviorState);
             this.disposableResources.Add(this.chainRepository);
-            this.disposableResources.Add(this.connectionManager);
             this.disposableResources.Add(this.nodeSettings.LoggerFactory);
```

Comment:
So after this change, what happens is that `BaseFeature.Stop` is called first and then `ConnectionManager.Dispose` is called. I am not sure that is correct because as you can see here, we dispose logger factory here. The problem with disposing logger factory is that once it is disposed, you must not call any constructor of any object that creates logger, because that will crash. So maybe we don't have any such thing, but if we don't, it is rather coincidence. 

The correct way, imho, is to dispose the logger factory as a last thing. So there should not be any major component running (such as connection manager) after it is disposed. 

I'm not saying this is the case currently, but imagine some component still runs after LF is disposed and a new message arrives from network or a new peer is connected, some behavior is created and the logger is needed ... 

So what we probably need here is to make sure the whole shutdown series is correct, which especially means that the components are stopped and disposed in correct order.

@dangershony can you see the problem here i've tried to described?

Reply:
yes I would even dispose of this tow components manually and not inside the list, they will always get disposed last, LF must be dispose after CM, in fact perhaps we dont even dispose of LF at all, as this will anyway get cleared away for shutdown.

@noescape00 regarding your comment yeah I agree we should control the processes of shutdown however the IDisposable pattern requires the method to be able to be called more then once without throwing.

But if the DI is calling Dispose then this is out of our control and perhaps we need to make changes (not use IDsiposable as this is a component that is completely under our control) and just clean it up at node shutdown.